dhis-2/dhis-e2e-test/src/test/java/org/hisp/dhis/tracker/importer/IdSchemeTests.java

```diff
@@ -0,0 +1,242 @@
+import org.hisp.dhis.actions.metadata.OrgUnitActions;
+import org.hisp.dhis.actions.metadata.ProgramActions;
+import org.hisp.dhis.actions.metadata.ProgramStageActions;
+import org.hisp.dhis.actions.metadata.RelationshipTypeActions;
+import org.hisp.dhis.actions.metadata.TrackedEntityAttributeActions;
+import org.hisp.dhis.actions.metadata.TrackedEntityTypeActions;
+import org.hisp.dhis.helpers.QueryParamsBuilder;
+import org.hisp.dhis.tracker.TrackerNtiApiTest;
+import org.hisp.dhis.tracker.importer.databuilder.EnrollmentDataBuilder;
+import org.hisp.dhis.tracker.importer.databuilder.EventDataBuilder;
+import org.hisp.dhis.tracker.importer.databuilder.RelationshipDataBuilder;
+import org.hisp.dhis.tracker.importer.databuilder.TeiDataBuilder;
+import org.hisp.dhis.utils.DataGenerator;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+import com.google.gson.JsonObject;
+
+/**
+ * @author Gintare Vilkelyte <vilkelyte.gintare@gmail.com>
+ */
+public class IdSchemeTests
+    extends TrackerNtiApiTest
+{
+    private final static String ATTRIBUTE_ID = "f3JrwRTeSSz";
+
+    @BeforeAll
+    public void beforeAll()
+    {
+        loginActions.loginAsAdmin();
+
+        new MetadataActions()
+            .importAndValidateMetadata( new File( "src/test/resources/tracker/idSchemesMetadata.json" ) );
+    }
+
+    private static Stream<Arguments> provideIdSchemeArguments()
+    {
+        return Stream.of(
+            Arguments.arguments( "CODE" ),
+            Arguments.arguments( "NAME" ),
+            Arguments.arguments( "ATTRIBUTE:" + ATTRIBUTE_ID ) );
+    }
+
+    @MethodSource( "provideIdSchemeArguments" )
+    @ParameterizedTest( name = "POST to /tracker?idScheme={0}" )
+    public void shouldImportTrackerProgramDataByIdScheme( String idScheme )
+    {
+        String teiId = new IdGenerator().generateUniqueId();
+
+        TestData data = new TestData( idScheme );
+
+        EventDataBuilder eventDataBuilder = new EventDataBuilder()
+            .setOu( data.getOrgUnit() )
+            .setProgramStage( data.getTrackerProgramStage() )
+            .addDataValue( data.getDataElement(), DataGenerator.randomString() )
+            .setAssignedUser( Constants.SUPER_USER_ID )
+            .setProgram( data.getTrackerProgram() );
+
+        EnrollmentDataBuilder enrollmentDataBuilder = new EnrollmentDataBuilder()
+            .setOu( data.getOrgUnit() )
+            .setProgram( data.getTrackerProgram() )
+            .addEvent( eventDataBuilder );
+
+        JsonObject payload = new TeiDataBuilder()
+            .setId( teiId )
+            .setTeiType( data.getTrackedEntityType() )
+            .setOu( data.getOrgUnit() )
+            .addAttribute( data.trackedEntityAttribute, DataGenerator.randomString() )
```

Comment:
Is there any reason you are not using the getters sometimes? like in `data.trackedEntityAttribute`

If not, I would keep it consistent 😄

Reply:
no reason, just clumsiness. Fixed now 👍